pkg/kubelet/dockertools/docker.go

```diff
@@ -306,8 +306,10 @@ func ConnectToDockerOrDie(dockerEndpoint string) DockerInterface {
 
 func milliCPUToShares(milliCPU int64) int64 {
 	if milliCPU == 0 {
-		// zero milliCPU means unset. Use kernel default.
-		return 0
+		// Docker converts zero milliCPU to unset, which maps to kernel default
+		// for unset: 1024. Return 2 here to really match kernel default for
+		// zero milliCPU.
+		return 2
```

Comment:
Should we log here so that if the user is not aware of the consequence of unset cpu limit, they can at least see it in the log (assuming we don't set the default in the apiserver).